src/coreclr/vm/methodtable.cpp

```diff
@@ -4317,7 +4317,10 @@ void MethodTable::DoFullyLoad(Generics::RecursionGraph * const pVisited,  const
         ClassLoader::ValidateMethodsWithCovariantReturnTypes(this);
     }
 
-    if ((level == CLASS_LOADED) && CORDisableJITOptimizations(this->GetModule()->GetDebuggerInfoBits()) && !HasInstantiation())
+    if ((level == CLASS_LOADED) && 
+        CORDisableJITOptimizations(this->GetModule()->GetDebuggerInfoBits()) &&
+        !HasInstantiation() &&
+        !GetModule()->GetAssembly()->IsLoading()) // Do not do this during the vtable fixup stage of C++/CLI assembly loading. See https://github.com/dotnet/runtime/issues/110365
```

Comment:
The comment is most appreciated. Two questions:

1) How do we know the "vtable fixup stage" is occurring? Is that `CLASS_LOADED`?
2) It seems like `IsLoading()` is implying that the type is being loaded _during_ assembly load (i.e., global statics are being initialized at that time?) and that combination is causing stress. This begs the question, is this limited to C++/CLI scenarios where statics for them are initialized at a time outside the ECMA loading process?

Reply:
Yes, these things are being loaded during assembly load. Technically this change will also impact scenarios where we have a module constructor, but really that's not a big deal either. This eager static's creation is really only here to make the debugger experience a bit prettier, so deferring that case too should be ok. For the loading stage.... this happens while the assembly load level is `FILE_LOAD_DELIVER_EVENTS` and is transitioning to `FILE_LOAD_VTABLE_FIXUPS`, but really we can run into issues any time the m_level is less than `FILE_LOADED`  This fix is a little aggresive as it doesn't just check for `FILE_LOADED` it checks for if its currently loading the assembly at all, but a slightly bigger fix is preferable here, as the diagnostic consequences of not filling this in are minimal, and the failure mode is so hard to debug.